hack/integration-test.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2023 Red Hat, Inc.
+#
+
+set -e
+
+INTEG_TEST_IMAGE=${INTEG_TEST_IMAGE:-"quay.io/kubevirt/builder:2306271234-e00d9fcf9"}
+PODMAN_SOCKET=${PODMAN_SOCKET:-"/run/podman/podman.sock"}
+
+detect_podman_socket() {
+    if curl --unix-socket "${PODMAN_SOCKET}" http://d/v3.0.0/libpod/info >/dev/null 2>&1; then
+        echo "${PODMAN_SOCKET}"
+    fi
+}
+
+if [ "${CONTAINER_CLIENT}" = "podman" ]; then
+    _cri_bin="podman --remote --url=unix://$(detect_podman_socket)"
+elif [ "${CONTAINER_CLIENT}" = "docker" ]; then
+    _cri_bin=docker
+else
+    _cri_socket=$(detect_podman_socket)
+    if [ -n "$_cri_socket" ]; then
+        _cri_bin="podman --remote --url=unix://$_cri_socket"
+        echo >&2 "selecting podman as container runtime"
+    elif docker ps >/dev/null 2>&1; then
+        _cri_bin=docker
+        echo >&2 "selecting docker as container runtime"
+    else
+        echo >&2 "no working container runtime found. Neither docker nor podman seems to work."
+        exit 1
+    fi
+fi
```

Comment:
I think you can source `hack/common.sh`, it detect the CRI and set KUBEVIRT_CRI.
It can later used as so `${KUBEVIRT_CRI} run ...`

Reply:
That bash script does many things that we do not need here.
I agree we should extract this logic to a common place, but this is not the focus of this PR. It is also not that trivial on how to arrange the scripts well, at the moment they seem to be very messy and consist a lot of things the callers do not really need.